spec.bs

```diff
@@ -880,3 +821,122 @@ specification adds another value for fenced frames:
 
 : "`fencedframe`"
 :: This [=navigable=] is displaying a <{fencedframe}>'s content
+
+<h3 id=csp-integration>Content Security Policy</h3>
+
+<h4 id=csp-intro>Intro</h4>
+
+*This section is non-normative*.
+
+Content Security Policy [[!CSP]] can ordinarily be used by web content associated with a
+{{Document}} that hosts a [=navigable container=] to limit the source of navigations in a [=child
+navigable=].
+
+In order to prevent [[!CSP]] from being used a communication side-channel exposing the [=URL=] of
+navigations inside a <{fencedframe}> to the site operating its embedder, the only [=source
+expressions=] that can influence a <{fencedframe}> navigation are:
+
+ * The <a grammar>scheme-source</a> "`https:`"
+ * The <a grammar>host-source</a> "`https://*:*`"
+ * The [=string=] "`*`"
+
+See our <a
+href=https://github.com/WICG/fenced-frame/blob/master/explainer/interaction_with_content_security_policy.md#proposal---scheme-source-matching>CSP
+explainer</a> that describes this.
+
+<h4 id=csp-algorithms>Algorithms</h4>
+
+<div algorithm=pre-request-check-csp>
+  Add a step after step 2 of the [=frame-src pre-request check=] that says:
+
+  3. If <var ignore>request</var>'s [=request/destination=] is "`fencedframe`", and this directive's
+     [=directive value|value=] does not [=set/contain=] either "`https:`", "`https://*:*`", or
+     "`*`", return "`Blocked`".
+</div>
+
+<div algorithm=post-request-check-csp>
+  Add a step after step 2 of the [=frame-src post-request check=] that says:
+
+  3. If <var ignore>request</var> [=request/destination=] is "`fencedframe`", and this directive's
+    [=directive value|value=] does not contain either "`https:`", "`https://*:*`", or "`*`", return
+    "`Blocked`".
+</div>
+
+Next, we modify the behavior of the CSPEE spec. If the embedding frame specifies a
+[=required csp=], fenced frames will not load. This is done to prevent arbitrary data flow from the
+embedder to the fenced frame.
+
+<div algorithm=cspee-changes>
+  Add a step after step 1 in the [=Is response to request blocked by context's required CSP?=]
+  algorithm that reads:
+
+  2. If <var ignore>context</var>'s [=required csp=] is not `null`, and <var ignore>request</var>
+    [=request/destination=] is "`fencedframe`", return "`Blocked`".
+</div>
+
+<h4 id=new-csp-directive>New fenced-frame-src [[!CSP]] [=directive=]</h4>
+
+Fenced frames are a different element from an iframe. Therefore, using the
```

Comment:
Somewhere in this section we'll need to modify https://w3c.github.io/webappsec-csp/#effective-directive-for-a-request right?